src/components/SiteSwitcher.tsx

```diff
@@ -2,28 +2,14 @@ import { Flex, Text, ButtonGroup, Button } from "@mantine/core"
 import { useAtom } from "jotai"
 import { Icon } from "@iconify/react"
 import cx from "classnames"
-import { useMutation } from "@tanstack/react-query"
 
 import { siteAtom } from "../store/site"
-import { switchSite } from "../utils/switch-site"
 
 import { SITES } from "../constants/sites"
 
-import { SiteKey } from "../types/site"
-
 export const SiteSwitcher = () => {
   const [currentSite, setCurrentSite] = useAtom(siteAtom)
 
-  const switchSiteMutation = useMutation({
-    mutationFn: switchSite,
-    mutationKey: ["login"],
-  })
-
-  async function changeSite(key: SiteKey) {
-    await switchSiteMutation.mutateAsync(key)
-    setCurrentSite(key)
-  }
```

Comment:
This is automatically handled by `@tanstack/react-query` when we provided the query key, e.g. `queryKey: ["categories", site]`